src/scripts/main.js

```diff
@@ -1,11 +1,25 @@
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve, reject) => {
+    document.addEventListener(eventName, (e) => {
+      if (e.target.id !== element.id) {
```

Comment:
The current logic checks if the event target's ID matches the element's ID. This might not work as expected if the event is triggered by a child element of the target element. Consider using `element.contains(e.target)` to check if the event target is the element itself or a descendant of it.